app/controllers/users_controller.rb

```diff
@@ -25,9 +26,27 @@ def create
     end
   end
 
+  def edit
+    @user = current_user
+  end
+
+  def update
+    @user = current_user
+    @user.update!(update_params)
+    flash[:success] = "Your data is updated"
+
+    redirect_to profile_path
+  end
+
   private
 
   def user_params
     params.require(:user).permit(:name, :street, :city, :state, :zipcode, :email, :password)
   end
+
+  def update_params
```

Comment:
@blake and @n-flint, since this always deletes the password, a user cannot ever change their password with this form.

Reply:
Corrected password edit page